### ISSUE TITLE
Coins included in Cashtokens spending doesn't respect "spend only fused coins" setting

### ISSUE THREAD

**Issue by imaginaryusername** [Electron-Cash/Electron-Cash#2799]
Right now Cashtokens spending (through Cashtokens tab -> spend tokens) seems to always just pick the largest non-frozen UTXO as its change and/or included extra BCH. It does not seem to follow usual coin chooser rules, and more problematically doesn't seem to respect "spend only fused coins" setting in the send tab. Custom manuevers like freezing is needed to get it to spend a fused coin.

Sending Cashtokens should either 1) check status of the "spend only fused coins" setting in send tab and respect that, or 2) get its own "spend only fused BCH" box. 

Additionally it'll be also nice to get it to follow normal EC coinchooser rules in general instead of always using the largest UTXO.

**Comment by imaginaryusername** [comment]
No, it happens when I don't include any extra BCH. Some BCH is always needed for change.

**Comment by fyookball** [comment]
so it includes a small amount of BCH as inputs,  in the tx but these are not taken from fused coins?

**Comment by imaginaryusername** [comment]
Yes, and it seems specifically looking for the _largest_ coins available regardless of fusion status. so if I have a 100bch utxo and a 0.001 bch utxo, it'll choose the 100bch consistently, shave off a bit for fees, and toss the 99.999... bch back as change.

**Comment by fyookball** [comment]
I was taking a look at this, we would need to add coinchooser logic inside the wallet make_token_send.  @cculianu told me its not necessarily a bug though.  My thought from looking at the code is if were' already building a transaction from est_tx_size() , maybe we could piggy back off that to get the "right" inputs, although I'm not 100% clear on how the existing logic works there so I would have to dig a bit deeper but if its not a bug, maybe I should hold off.

**Comment by imaginaryusername** [comment]
It is absolutely a bug, taking the largest coin is not right - it prevents the large coin from otherwise being used in more useful contexts with its privacy intact (if fused) or confirmation intact (if on bitpay etc.) Picking a 5bch coin to pay 0.00001 in fees when other coins are available is just overall terrible. 

Yes, using the regular BCH send's coinchooser will give you much better coins to use, that's the idea here. I don't know how easy/hard it would be to import that logic though.

**Comment by cculianu** [comment]
I was joking to Jonald that it's not a bug it's a misfeature. Philosophical argument here -- I mean, it's just a suboptimal feature is all not technically a bug. Like I never intended it to pick the right coin when I threw this together in 1-2 weeks in May 2023. But yes, it should be corrected.
